### PR TITLE
RATEST-324:RefApp2.x Service

### PR DESCRIPTION
**Pull request by sherrif10**
https://issues.openmrs.org/browse/RATEST-324 cc @kdaud . This seemed to the last ticket standing so i decided to quicky push it here.

### CODE REVIEW

File: .github/workflows/refapp-2x-service.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+name: RefApp 2.x Service
+on:
+  push:
+    branches: [master]
+  pull_request:
+    branches: [master]
+    paths:
+      - "qaframework-bdd-tests/src/**/*.*"
+      - "qaframework-bdd-tests/package.json"
```
Comment (kdaud) [inline .github/workflows/refapp-2x-service.yml:9]: Add the path for the pom.xml that resides in qaframework-bdd-tests folder to have the test run on PR suggesting library upgrades

Comment (sherrif10) [reply .github/workflows/refapp-2x-service.yml:9]: Hello @kdaud  am still little confused about this change . Probably lets first handle your case and i will come back to this one as soon as it make sense to me.

File: qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/ServiceSteps.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+
+import org.apache.commons.lang3.RandomStringUtils;
+import org.openmrs.contrib.qaframework.RunTest;
+import org.openmrs.contrib.qaframework.page.AppointmentSchedulingPage;
+import org.openmrs.contrib.qaframework.page.ManageServiceTypesPage;
+import org.openmrs.contrib.qaframework.page.ServicePage;
+
+import io.cucumber.java.Before;
```
Comment (kdaud) [inline qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/ServiceSteps.java:12]: Align the imports with the existing import order. Cross check the existing file and align

**Comment by sherrif10** [comment]
@kdaud  Is this ready for merge

**Comment by kdaud** [comment]
What plan do you have for the equivalent test present in the legacy sub module?

File: qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/ServiceSteps.java [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        servicePage.setDescription(description);
+    }
+
+    @And("user clicks on save button")
+    public void userClicksOnSave() {
+        manageServiceTypesPage = servicePage.save();
+    }
+
+    @And("system checks the saved service")
+    public void loadManageServicePage() {
```
Comment (kdaud) [inline qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/ServiceSteps.java:64]: The method name is not align with the user step.

File: qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/ServiceSteps.java [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    private AppointmentSchedulingPage appointmentSchedulingPage;
+    private ManageServiceTypesPage manageServiceTypesPage;
+    private ServicePage servicePage;
+    
+    private String name;
+    private String duration;
+    private String description;
+    
+    @Before(RunTest.HOOK.SELENIUM_SERVICE)
+    public void setup() throws Exception {
```
Comment (kdaud) [inline qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/ServiceSteps.java:29]: `setup()`?

File: qaframework-bdd-tests/src/test/resources/features/refapp-2.x/17-Service/service.feature [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    And system checks the saved service
+    #edit section
+    And user clicks on edit service type icon
+    And user clicks again on save button
+    And system checks the editted service type
+    #delete section
+    And user clicks on delete service type icon
+    And user clicks on comfirm button
+    Then user checks the availability of service type by name
+
```
Comment (kdaud) [inline qaframework-bdd-tests/src/test/resources/features/refapp-2.x/17-Service/service.feature:22]: Isn't one empty line enough to cater for git collaborators?

Comment (sherrif10) [reply qaframework-bdd-tests/src/test/resources/features/refapp-2.x/17-Service/service.feature]: Resolved

File: qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/ServiceSteps.java [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+    @Before(RunTest.HOOK.SELENIUM_SERVICE)
+    public void setUp() throws Exception {
+        initiateWithLogin();
+        name = RandomStringUtils.randomAlphabetic(5);
+        duration = RandomStringUtils.randomNumeric(2);
+        description = RandomStringUtils.randomAlphabetic(10);
+    }
+
+    @After(RunTest.HOOK.SELENIUM_SERVICE)
+    public void destroy(){
```
Comment (kdaud) [inline qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/ServiceSteps.java:38]:
```suggestion
    public void destroy() {
```

File: qaframework-bdd-tests/src/test/resources/features/refapp-2.x/17-Service/service.feature [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+    When user goes to manage service page
+    Then user clicks on new service page
+    And user enters service details
+
+  @selenium
+  @service
+  Scenario: Service management
+    And user clicks on save button
+    And system checks the saved service
+    #edit section
```
Comment (kdaud) [inline qaframework-bdd-tests/src/test/resources/features/refapp-2.x/17-Service/service.feature:14]: What do you think about `#edit section `vs `#edit service type`

Comment (sherrif10) [reply qaframework-bdd-tests/src/test/resources/features/refapp-2.x/17-Service/service.feature]: Resolved.

File: qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/ServiceSteps.java [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+
+    @And("system checks the saved service")
+    public void systemChecksSavedService() {
+        assertThat(manageServiceTypesPage.getServiceType(name), is(true));
+    }
+
+    @And("user clicks on edit service type icon")
+    public void userClicksOnEditServiceType() {
+        servicePage = manageServiceTypesPage.editServiceType(name);
+        servicePage.setName(name + name);
```
[the remark this answers is not in the document]
Comment (kdaud) [reply qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/ServiceSteps.java:77]: Take a look at what the gherkin step says and what is being implemented at line `77`

[the remark this answers is not in the document]
Comment (kdaud) [reply qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/ServiceSteps.java:77]: Can you reflect line 77 in the gherkin step as well?

**Review by kdaud — APPROVED** [review]

**Pull request merged**